projects/cm_mcu/LocalTasks.c

```diff
@@ -335,20 +339,87 @@ struct MonitorI2CTaskArgs_t clockr0a_args = {
     .stack_size = 4096U,
 };
 
+void setFFmask(uint32_t ff_combined_mask)
+{
+  uint32_t whichff = 0;
+  uint32_t data = 0;
+
+  log_info(LOG_SERVICE, "Setting a bit mask of enabled Fireflys to 1 \r\n");
+
+  for (; whichff < NFIREFLIES; ++whichff) {
+    uint8_t val = (ff_combined_mask >> (whichff)) & 0x01;
+    log_debug(LOG_SERVICE, "%17s: %x", ff_moni2c_addrs[whichff].name, val);
+    data += (!val << whichff);
+  }
+
+  ff_USER_mask = read_eeprom_single(EEPROM_ID_FF_ADDR);
+  ff_PRESENT_mask = data;
+  uint64_t block = EEPROMBlockFromAddr(ADDR_FF);
+
+  uint64_t unlock = EPRMMessage((uint64_t)EPRM_UNLOCK_BLOCK, block, PASS);
+  xQueueSendToBack(xEPRMQueue_in, &unlock, portMAX_DELAY);
+
+  uint64_t message = EPRMMessage((uint64_t)EPRM_WRITE_SINGLE, ADDR_FF, data);
+  xQueueSendToBack(xEPRMQueue_in, &message, portMAX_DELAY);
+
+  uint64_t lock = EPRMMessage((uint64_t)EPRM_LOCK_BLOCK, block << 32, 0);
+  xQueueSendToBack(xEPRMQueue_in, &lock, portMAX_DELAY);
+
+  return;
+}
+
+void readFFpresent()
+{
+  while (xSemaphoreTake(i2c4_sem, (TickType_t)10) == pdFALSE)
+    ;
+
+  uint32_t present_FFLDAQ_F1, present_FFL12_F1, present_FFLDAQ_F2, present_FFL12_F2;
+  // # set first I2C switch on channel 4 (U14, address 0x70) to port 7
+  apollo_i2c_ctl_w(4, 0x70, 1, 0x80);
+  apollo_i2c_ctl_reg_r(4, 0x20, 1, 0x01, 1, &present_FFL12_F1);
+
+  apollo_i2c_ctl_w(4, 0x71, 1, 0x40);
+  apollo_i2c_ctl_reg_r(4, 0x21, 1, 0x00, 1, &present_FFLDAQ_F1);
+
+  xSemaphoreGive(i2c4_sem); // if we have a semaphore, give it
+
+  while (xSemaphoreTake(i2c3_sem, (TickType_t)10) == pdFALSE)
+    ;
+  apollo_i2c_ctl_w(3, 0x70, 1, 0x80);
+  apollo_i2c_ctl_reg_r(3, 0x20, 1, 0x01, 1, &present_FFL12_F2);
+
+  apollo_i2c_ctl_w(3, 0x71, 1, 0x40);
+  apollo_i2c_ctl_reg_r(3, 0x21, 1, 0x00, 1, &present_FFLDAQ_F2);
+
+  xSemaphoreGive(i2c3_sem); // if we have a semaphore, give it
+
+  present_FFL12_F1 = present_FFL12_F1 & 0x3FU;         // bottom 6 bits
+  present_FFL12_F2 = present_FFL12_F2 & 0x3FU;         // bottom 6 bits
+  present_FFLDAQ_F1 = (present_FFLDAQ_F1 >> 4) & 0xFU; // bits 4-7
+  present_FFLDAQ_F2 = (present_FFLDAQ_F2 >> 4) & 0xFU; // bits 4-7
+
+  uint32_t ff_combined_mask = ((present_FFLDAQ_F2) << 16) | // 4 bits
+                              ((present_FFL12_F2) << 10) |  // 6 bits
+                              (present_FFLDAQ_F1) << 6 |    // 4 bits
+                              ((present_FFL12_F1));         // 6 bits
+
+  setFFmask(ff_combined_mask);
+}
+
 bool isEnabledFF(int ff)
 {
   // firefly config stored in on-board EEPROM
   static bool configured = false;
```

Comment:
Once you deleted the storage of the mask in this function as a static variable, the static bool here and setting it no longer makes any sense and shoudl be removed....

Reply:
it's removed now. Done.